libraries/AP_Compass/AP_Compass_MAG3110.cpp

```diff
@@ -251,7 +255,7 @@ void AP_Compass_MAG3110::read()
         return;
     }
 
-    Vector3f field(_mag_x_accum, _mag_y_accum, _mag_z_accum);
+    Vector3f field = Vector3f(_mag_x_accum, _mag_y_accum, _mag_z_accum);
```

Comment:
this shouldn't matter, but why did you change to the copy constructor ?

Reply:
I got a garbage in 'field' variable before somehow (in debugger). This compiles to pure calculations without any memory copy